src/components/ReportsPage/HeatmapPage/HeatmapPage.tsx

```diff
@@ -56,6 +77,40 @@ export const HeatmapPage = () => {
     return buildUrl(queryParamsObj, location.pathname);
   }, [location, selectedScorecardId, groupBy, headerType]);
 
+  const scorecardsResult = useCortexApi(api => api.getScorecards());
+
+  const selectedScoreCard = useMemo(() => {
+    return scorecardsResult.value?.find((scorecard) => scorecard.id === selectedScorecardId)
+  }, [scorecardsResult, selectedScorecardId]);
+
+  const entityCategory = useMemo(() => {
+    return getEntityCategoryFromFilter(selectedScoreCard?.filter) ?? 'Entity';
+  }, [selectedScoreCard]);
+
+  const excludedGroupBys = useMemo(() => {
```

Comment:
It looks like those three useMemos are tightly coupled. What do you think about combining them into single useMemo, like:

```
  const { entityCategory, excludedGroupBys } = useMemo(() => {
    const selectedScorecard = scorecardsResult.value?.find((scorecard) => scorecard.id === selectedScorecardId)
    
    return {
      entityCategory: getEntityCategoryFromFilter(selectedScorecard?.filter) ?? 'Entity',
      excludedGroupBys: teamBasedCardSelected(selectedScoreCard) ? [GroupByOption.TEAM] : []
    };
  }, [scorecardsResult, selectedScorecardId]);
```